ipaserver/install/custodiainstance.py

```diff
@@ -122,13 +124,34 @@ def import_dm_password(self, master_host_name):
         cli = self.__CustodiaClient(server=master_host_name)
         cli.fetch_key('dm/DMHash')
 
+    def __wait_keys(self, host, timeout=300):
+        ldap_uri = 'ldap://%s' % host
+        principal = 'host/%s@%s' % (self.fqdn, self.realm)
+        deadline = int(time.time()) + timeout
+        root_logger.info("Waiting up to {} seconds to see our keys "
+                         "appear on host: {}".format(timeout, host))
+
+        konn = KEMLdap(ldap_uri)
+        while True:
+            try:
+                konn.get_key(KEY_USAGE_ENC, principal)
+                return
+            except Exception:
```

Comment:
I don't suggest not waiting, of course. I would have gone for just `ValueError` but you're probably right that the connection error might appear.
It might be worth to do this, at least:
```python
exc = None
while int(time.time()) < deadline:
    try:
        konn.get_key(KEY_USAGE_ENC, principal)
        break
    except ValueError as e:
        exc = e
        time.sleep(1)
    except Exception as e:
        root_logger.debug("Failed to get keys: '{err}'".format(err=e))
        exc = e
        time.sleep(1)
else:
    if exc is not None:
        raise exc
    else:
        raise RuntimeError("Unable to obtain keys in expected time.")
```